### PR TITLE
Handle key and secret permissions errors

### PR DESCRIPTION
**Pull request by Gonzalo-Avalos-Ribas**
"RestError: AKV10032: Invalid issuer. Expected one of https://sts.windows.net/5ae1af62-9505-4097-a69a-c1553ef7840e/, https://sts.windows.net/f8cdef31-a31e-4b4a-93e4-5f571e91255a/, https://sts.windows.net/e2d54eb5-3869-4f70-8578-dee5fc7331f4/, found https://sts.windows.net/e5385570-36f1-4d50-aaf8-db2f87aa884a/. \n[...]"

We are requesting the keys with the wrong tenant/application. All of our integrations are single tenant based, so it would be impossible to ingest!

### CODE REVIEW

File: src/steps/resource-manager/key-vault/client.ts [hunk 1]
```diff
@@ -54,6 +54,21 @@ export class KeyVaultClient extends Client {
...
+        //We could have cases where an AD has the keyvault but we need to access the keys
+        //using another tenant. For now, lets just skip this cases.
+        this.logger.warn(
+          { err: err, vaultUri: vaultUri },
+          'Failed to retrieve a VaultKey',
+        );
+        this.logger.publishWarnEvent({
+          name: IntegrationWarnEventName.MissingEntity,
+          description: `This tenant/application is not allowed to access keys for vault ${vaultUri}`,
+        });
```
Comment (zemberdotnet) [inline src/steps/resource-manager/key-vault/client.ts:71]: If line 44 throws, does it continue the loop or does it end the loop? Are we skipping just one or ending the whole step?

Comment (Gonzalo-Avalos-Ribas) [reply src/steps/resource-manager/key-vault/client.ts:71]: It's ending the loop ! we are skipping all the keys or secrets for that keyvault. 
But we continue iterating other vaults

Comment (zemberdotnet) [reply src/steps/resource-manager/key-vault/client.ts:71]: I see. So this logs message only prints once. LGTM 👍

**Review by zemberdotnet — APPROVED** [review]

File: src/steps/resource-manager/key-vault/client.ts [hunk 2]
```diff
@@ -54,6 +54,21 @@ export class KeyVaultClient extends Client {
           name: IntegrationWarnEventName.MissingPermission,
           description: `Missing a Key Vault access policy. A Key Vault access policy determines whether a given security principal can perform different operations on Key Vault secrets, keys and certificates. Please follow the steps outlined here https://go.microsoft.com/fwlink/?linkid=2125287 and assign a "list" key permission in order to fetch these keys for your Key Vault ${vaultUri}.`,
         });
+      } else if (
+        err.statusCode === 401 &&
+        err.message.toString().includes('AKV10032')
```
Comment (zemberdotnet) [inline src/steps/resource-manager/key-vault/client.ts:59]: What does this error code mean?

**Pull request merged**